modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -926,14 +960,28 @@ public CompletableFuture<Void> stopAsync(ComponentContext componentContext) {
         shutdownAndAwaitTermination(executor, shutdownTimeoutSeconds, TimeUnit.SECONDS);
         shutdownAndAwaitTermination(replicasCreationExecutor, shutdownTimeoutSeconds, TimeUnit.SECONDS);
 
+        try {
+            IgniteUtils.closeAllManually(
+                    () -> CompletableFuture.allOf(raftClientsFutures.values().toArray(new CompletableFuture<?>[0]))
+                            .get(shutdownTimeoutSeconds, TimeUnit.SECONDS),
+                    () -> raftClientsFutures.forEach((raftNodeId, raftClientFuture) -> {
+                        try {
+                            raftManager.stopRaftNode(raftNodeId);
+                        } catch (NodeStoppingException e) {
+                            throw new AssertionError("Raft node is stopping [raftNodeId=" + raftNodeId
+                                    + "], but it's abnormal, because Raft Manager must stop strictly after Replica Manager.", e);
+                        }
+                    }),
+                    () -> replicas.values().forEach(replicaFuture -> replicaFuture.completeExceptionally(new NodeStoppingException()))
+            );
+        } catch (Exception e) {
+            LOG.error("There are exceptions during Replica Manager stop", e);
```

Comment:
It's stopAsync, thus it's better to return failedFuture(e);

Reply:
Done!